var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -37,7 +38,16 @@ class Geant4(CMakePackage):
             default=_cxxstd_values[0],
             values=_cxxstd_values,
             multi=False,
-            description='Use the specified C++ standard when building.')
+            description='Use the specified C++ standard when building.',
+            when='@10')
+    variant('cxxstd',
+            default=_cxxstd_values[2],
+            values=_cxxstd_values,
+            multi=False,
+            description='Use the specified C++ standard when building.',
+            when='@11')
+    conflicts('cxxstd=11', when='@11:', msg='geant4@11: only supports cxxstd=17')
+    conflicts('cxxstd=14', when='@11:', msg='geant4@11: only supports cxxstd=17')
```

Comment:
I was hoping for something like
```
    _cxxstd_values = ('11', '14', '17', '20', '23')
    variant('cxxstd',
            default=_cxxstd_values[0],
            values=_cxxstd_values[0:3],
            multi=False,
            description='Use the specified C++ standard when building.',
            when='@10')
    variant('cxxstd',
            default=_cxxstd_values[2],
            values=_cxxstd_values[2:],
            multi=False,
            description='Use the specified C++ standard when building.',
            when='@11')
```
which would avoid the lists of conflicts, but:
- it doesn't work with the `for std in _cxxstd_values:` below, where a fix is more lines than than 2,
- not all dependencies have cxxstd 20 support (xerces, boost, vecgeom).
So, for now, we probably have to stop at 17 and work on other packages to get cxxstd 20 support. And maybe something like `conflicts('cxxstd<17', when='@11:')`.

Reply:
I think this is not the right use case for a multi valued variant. The new concretizer can automatically change cxxstd with the conflict statements.

Reply:
Not sure I understand if you are suggesting a change or if your comment is in response to my comment only. Can you clarify?

Reply:
Yeah suggesting a change. You need to remove the duplicate `variant` and the `when`. Leaving the conflicts should allow the variant to automatically change.

Reply:
I see. The concretizer is smarter than me :-)